services/provider/server/server.go

```diff
@@ -213,32 +213,103 @@ func (s *OCSProviderServer) GetStorageConfig(ctx context.Context, req *pb.Storag
+
+		for _, kubeResource := range kubeResources {
+			gvk, err := apiutil.GVKForObject(kubeResource, s.scheme)
+			if err != nil {
+				return nil, err
+			}
+			switch gvk.Kind {
+			case "CephConnection":
+				cephConn := kubeResource.(*csiopv1a1.CephConnection)
+				response.ExternalResource = append(
+					response.ExternalResource,
+					&pb.ExternalResource{
+						Name: cephConn.Name,
+						Kind: "CephConnection",
+						Data: mustMarshal(cephConn.Spec),
+					},
+				)
+			case "ClusterResourceQuota":
+				quota := kubeResource.(*quotav1.ClusterResourceQuota)
+				response.ExternalResource = append(
+					response.ExternalResource,
+					&pb.ExternalResource{
+						Name: quota.Name,
+						Kind: "ClusterResourceQuota",
+						Data: mustMarshal(quota.Spec),
+					},
+				)
+			case "ClientProfileMapping":
+				clientProfileMapping := kubeResource.(*csiopv1a1.ClientProfileMapping)
+				response.ExternalResource = append(
+					response.ExternalResource,
+					&pb.ExternalResource{
+						Name: clientProfileMapping.Name,
+						Kind: "ClientProfileMapping",
+						Data: mustMarshal(clientProfileMapping.Spec),
+					},
+				)
+			case "Noobaa":
+				noobaa := kubeResource.(*nbv1.NooBaa)
+				response.ExternalResource = append(
+					response.ExternalResource,
+					&pb.ExternalResource{
+						Name: noobaa.Name,
+						Kind: "Noobaa",
+						Data: mustMarshal(noobaa.Spec),
+					},
+				)
+			case "Secret":
+				secret := kubeResource.(*corev1.Secret)
+				if secret.Name == "noobaa-remote-join-secret" {
+					oldSecretFormat := map[string]string{}
+					for k, v := range secret.Data {
+						oldSecretFormat[k] = string(v)
+					}
+					response.ExternalResource = append(
+						response.ExternalResource,
+						&pb.ExternalResource{
+							Name: secret.Name,
+							Kind: "Secret",
+							Data: mustMarshal(oldSecretFormat),
+						},
+					)
+				}
+			}
 
-		storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
-		if err != nil {
-			return nil, err
 		}
 
 		inMaintenanceMode, err := s.isSystemInMaintenanceMode(ctx)
 		if err != nil {
 			klog.Error(err)
 			return nil, status.Errorf(codes.Internal, "Failed to get maintenance mode status.")
 		}
+		response.SystemAttributes.SystemInMaintenanceMode = inMaintenanceMode
 
 		isConsumerMirrorEnabled, err := s.isConsumerMirrorEnabled(ctx, consumerObj)
 		if err != nil {
 			klog.Error(err)
 			return nil, status.Errorf(codes.Internal, "Failed to get mirroring status for consumer.")
 		}
+		response.SystemAttributes.MirrorEnabled = isConsumerMirrorEnabled
+
+		storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+		if err != nil {
+			return nil, err
+		}
+
+		channelName, err := s.getOCSSubscriptionChannel(ctx)
+		if err != nil {
+			klog.Errorf("failed to get channel name for Client Operator: %v", err)
+			return nil, status.Errorf(codes.Internal, "failed to get channel name for Client Operator")
+		}
```

Comment:
Is there a reason to add these. the 4.18 client don't do anything with these

Reply:
This is required for the hash we send from GetStorageInfo